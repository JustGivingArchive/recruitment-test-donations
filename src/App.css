body {
    padding: 0;
}
.app-container {
    background: blue;
    margin: 0 auto;
    display: flex;
    flex-direction: column;
    align-items: center;
}
