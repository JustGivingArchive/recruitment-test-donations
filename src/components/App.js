import React from 'react';
import Charity from './charity/Charity.js';

const App = () => {
  return (
    <div>
      <h1>Fundraising!</h1>
      <Charity />
    </div>
  );
};

export default App;
