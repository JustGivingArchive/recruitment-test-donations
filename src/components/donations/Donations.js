import React from 'react';
import IndividualDonation from './IndividualDonation.js';

// donations are served from the API as an array, so we can map over them
// and create a donation component for each

const Donations = ({ donations }) => {
  return (
    <div className="content_container">
      {donations.map((data, key) => <IndividualDonation
        key={key}
        {...data} />)}
    </div>
  );
};

export default Donations;
