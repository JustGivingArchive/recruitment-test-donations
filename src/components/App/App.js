import React from "react";
import ReactDOM from "react-dom";
import Header from '../Header/Header'
import Main from '../Main/Main'
import Footer from '../Footer/Footer'

import './App.css'

export const App = () => {
  return (
    <div className="app-container">
      <Header />
        <Main />
      <Footer/>
    </div>
    
  )
};
