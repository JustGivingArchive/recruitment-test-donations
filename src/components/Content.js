import React from 'react';
import Charity from './charity/Charity.js';
import Donations from './donations/Donations.js';
import CharitySelector from './charity/Charityselector.js';

const Content = (props) => {
  console.log(props);
  if (props.dropDown) {
    return (
      <CharitySelector select={props.select} />
    )
  }

  if (props.charityLoaded && props.donationsLoaded) {
    return (
      <div>
        <Charity info={props.charityInfo} />
        <Donations donations={props.donations} />
      </div>
    );
  }

  return (
      <div className="main_container">
        <h2 className="subtitle">Loading</h2>
      </div>
  );
};

export default Content;
