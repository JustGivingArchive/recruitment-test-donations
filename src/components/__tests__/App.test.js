import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from "enzyme";

import App from '../App.js';
import Donations from '../donations/Donations.js';
import Content from '../Content.js';

describe('App', () => {
  let props;
  let mounted;
  const app = () => {
    if (!mounted) {
      mounted = mount(
        <App {...props} />
      );
    }
    return mounted;
  }

  beforeEach(() => {
    props = {
    };
    mounted = undefined;
  });

  it('matches snapshot', () => {
    const tree = renderer.create(
      <App />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Always renders a containing div', () => {
    const divs = app().find("div");
    expect(divs.length).toBeGreaterThan(0);
  });

  it("wrapping div contains everything", () => {
    const divs = app().find("div");
    const wrappingDiv = divs.first();
    expect(wrappingDiv.children()).toEqual(app().children());
  });

  it("always renders Content", () => {
    expect(app().find(Content).length).toBe(1);
  })
});
