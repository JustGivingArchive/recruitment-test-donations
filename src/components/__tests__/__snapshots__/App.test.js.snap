exports[`App renders correctly 1`] = `
<div>
  <h1>
    Fundraising!
  </h1>
  <div
    class="charity_container">
    <h2>
      Charity Name
    </h2>
    <p>
      Charity description
    </p>
    <img
      alt="logo of charity name"
      src="logo-url" />
  </div>
</div>
`;
